include/sys/atexit.h

```diff
@@ -0,0 +1,81 @@
+/****************************************************************************
+ * include/sys/atexit.h
```

Comment:
it's not good to add a non standard header file in include/sys folder and is better to define atexit related type and help function to tls.h.

Reply:
I do think this belongs in either include/sys or somewhere in libc / stdlib. These definitions are needed by stdlib functions, TLS is just a storage for `struct atexit_list_s`.

Reply:
Yes, tls is just a storage, but which require the full definition to know the space to be resolved, even you define these types and functions in some header file, you still need include that file in tls.h. So, I propose tls.h, but you may find better place.

Reply:
I will try to figure out something tomorrow. e.g. glibc puts this in stdlib/exit.h, but I don't think we have header visibility to that folder (-I flags do not include stdlib/) ?

Reply:
Now that I think of it, maybe tls.h is good enough. That is the "re-entrancy" structure we use in NuttX right ? So semantically this stuff can go there.

Reply:
@xiaoxiang781216 I moved the header to include/nuttx, there seem to be quite many implementation specific headers there so that seemed like a good place